src/app/index.js

```diff
@@ -1,69 +1,208 @@
 }
 
+/**============= Slider show functionality on home page =================== */
 function showSlides(n) {
-  var i;
-  if (n > slides.length) { this.slideIndex = 1; }    
-  if (n < 1) {this.slideIndex = slides.length}
+  let i;
+  if (n > slides.length) { slideIndex = 1; }    
+  if (n < 1) {slideIndex = slides.length}
   for (i = 0; i < slides.length; i++) {
     slides[i].style.display = "none";  
   }
   for (i = 0; i < dots.length; i++) {
-      dots[i].className = dots[i].className.replace(" active", "");
+    dots[i].className = dots[i].className.replace(" active", "");
   }
-  slides[this.slideIndex-1].style.display = "block";  
-  dots[this.slideIndex-1].className += " active";
+  slides[slideIndex-1].style.display = "block";  
+  dots[slideIndex-1].className += " active";
 }
 
-var routes = {
-    'signup': signup,
-    'signin': login,
-    'home': home,
-    'products': products
+function onInitSlider() {
+  // prev and next slider button
+  $('.prev').on('click', function() {
+    plusSlide(-1);
+  });
+
+  $('.next').on('click', function() {
+    plusSlide(1);
+  });
+
+  const bannerList = banners.banners;
+  bannerList.forEach(function(banner, index) {
+    $(`.dot-${index}`).on('click', function() {
+      currentSlide(index);
+    });
+  })
 }
 
-function onNavItemClick(section) {
-  content.innerHTML = routes[section];
-  if(section === 'home') {
-    this.slideIndex = 1;
-    this.showSlides(this.slideIndex);
-  }
+function currentSlide(n) {
+  showSlides(slideIndex = n);
 }
 
-/**============= Login Input =================== */
-(function ($) {
-    "use strict";
-    /*==================================================================
-    [ Focus input ]*/
-    $('.input100').each(function(){
-        $(this).on('blur', function(){
-            if($(this).val().trim() != "") {
-                $(this).addClass('has-val');
-            }
-            else {
-                $(this).removeClass('has-val');
-            }
-        })    
-    });
+function plusSlide(n) {
+  showSlides(slideIndex += n);
+}
 
-    var showPass = 0;
-    $('.btn-show-pass').on('click', function(){
-        if(showPass == 0) {
-            $(this).next('input').attr('type','text');
-            showPass = 1;
-        }
-        else {
-            $(this).next('input').attr('type','password');
-            showPass = 0;
-        }
+function onInitCategories(categories) {
+
+  const allCategories = categories.categories;
+
+  // on mobile screen select is visible
+  $('select').on('change', function() {
+    onCategorySelection(this.value);
+  });
+
+  // on tablet and desktop screen
+  allCategories.forEach((category, index) =>  {
+    $(`.item-${index}`).on('click', function(el) {
+      onCategorySelection(el.target.innerText);
     });
+  });
+}
+
+function onCategorySelection(selectedCategory) {
+  const allCategories = categories.categories;
+  const selectedCategoryDetail = allCategories.filter(item => item.name === selectedCategory);
```

Comment:
Why can't you figure the category out from DOM element, these are unnecessary loops you are running